core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -535,10 +535,8 @@ class DAGScheduler(
     jobIdToActiveJob -= job.jobId
     activeJobs -= job
     job.finalStage match {
-      case r: ResultStage =>
-        r.resultOfJob = None
-      case m: ShuffleMapStage =>
-        m.mapStageJobs = m.mapStageJobs.filter(_ != job)
+      case r: ResultStage => r.removeActiveJob()
```

Comment:
one thing i'm thinking is whether we should move "removeActiveJob" and "addActiveJob" to Stage, and have both ResultStage and ShuffleMapStage implement those.